src/net/http/DownloadManager.cpp

```
@@ -87,47 +88,80 @@ class DownloadManagerThread final
     });
-
-    if (!task)
-      Start();
   }
 
   void Cancel(Path relative_path) noexcept {
-    auto i = std::find(queue.begin(), queue.end(), relative_path);
-    if (i == queue.end())
-      return;
+    bool cancel_active = false;
+    bool cancelled = false;
```

Cppcheck (2.21.0)

[error] Rethrowing current exception with 'throw;', it seems there is no current exception to rethrow. If there is no current exception this calls std

(rethrowNoCurrentException)